tests/oeedger8r/host/testdeepcopy.cpp

```diff
@@ -47,6 +47,414 @@ std::array<T, 2> init_structs()
     return s;
 }
 
+// Assert that the struct is deep-copied such that `s->ptr` has a copy
+// of `s->count` elements of `data`.
+void ocall_deepcopy_countparam(CountParamStruct* s)
+{
+    OE_TEST(s->count == 7);
+    OE_TEST(s->size == 64);
+    for (size_t i = 0; i < s->count; ++i)
+        OE_TEST(s->ptr[i] == data[i]);
+
+    // Modify the member used by the size attribute, which should
+    // not affect the value on the caller side.
+    s->count = 5;
+    // Modifiy the member not used by the size/count attributes,
+    // which should affect the value on the caller side.
+    s->size = 200;
+}
+
+void ocall_deepcopy_countparam_return_large(CountParamStruct* s)
+{
+    OE_TEST(s->count == 7);
+    OE_TEST(s->size == 64);
+    for (size_t i = 0; i < s->count; ++i)
+        OE_TEST(s->ptr[i] == data[i]);
+
+    // Set the value to the member used by the count attribute larger
+    // than the supplied value. Expect to fail.
+    s->count = 100;
+    // Modifiy the member not used by the size/count attributes,
```

Comment:
```suggestion
    // Modify the member not used by the size/count attributes,
```
nit: please fix spelling throughout this file